### PR TITLE
clean status conditions

### PR DESCRIPTION
**Pull request by jkhelil**

### CODE REVIEW

File: api/v1alpha2/jenkins_types.go [hunk 1]
```diff
@@ -470,13 +471,11 @@ type JenkinsStatus struct {
...
 
-	// Phase is a simple, high-level summary of where the Jenkins is in its lifecycle.
-	// There are five possible phase values:
-	// Pending: The Jenkins has been accepted by the Kubernetes system, but one or more of the required resources have not been created.
-	// Available: All of the resources for the Jenkins are ready.
-	// Failed: At least one resource has experienced a failure.
-	// Unknown: For some reason the state of the Jenkins phase could not be obtained.
-	Phase string `json:"phase"`
+	// Conditions describes the state of the jenkins resource.
+	// +patchMergeKey=type
```
Comment (akram) [inline api/v1alpha2/jenkins_types.go:477]: are we really using these ones? what is it for?

File: config/crd/bases/jenkins.jenkins.io_jenkins.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.3.0
+    controller-gen.kubebuilder.io/version: v0.2.5
```
Comment (akram) [inline config/crd/bases/jenkins.jenkins.io_jenkins.yaml]: I think you must upgrade your kubebuilder version

**Review by akram — CHANGES_REQUESTED** [review]

File: controllers/jenkins_controller.go [hunk 3]
```diff
@@ -57,6 +57,12 @@ const (
 	ConfigMapKind          = "ConfigMap"
 	containerProbeURI      = "login"
 	containerProbePortName = "http"
+
+	reconcileInit             = "Init"
+	reconcileInitMessage      = "Initializing HyperConverged cluster"
```
Comment (akram) [inline controllers/jenkins_controller.go]: Please change this to remove any reference to HyperConverged / hco . Replace that with `Jenkins Instance` or `Jenkins Controller` or `Jenkins Server` (Jenkins Master wording being deprecated )

File: controllers/jenkins_controller.go [hunk 4]
```diff
@@ -223,37 +235,27 @@ func (r *JenkinsReconciler) reconcile(ctx context.Context, request ctrl.Request,
...
-		err = r.Status().Update(ctx, jenkins)
-		if err != nil {
-			logger.Error(err, "Failed to update Jenkins status")
-			return ctrl.Result{}, err
-		}
-
-		logger.Info("Reconcile loop success !!!")
-		logger.Info(message)
+	jenkins.conditions.setStatusCondition(conditionsv1.Condition{
+		Type:    ConditionReconcileComplete,
```
Comment (akram) [inline controllers/jenkins_controller.go:254]: can we rename it to `ReconciliationComplete`?

File: controllers/jenkins_controller.go [hunk 5]
```diff
@@ -477,3 +479,36 @@ func basePlugins() (result []v1alpha2.Plugin) {
 func (r *JenkinsReconciler) isJenkinsPodTerminating(err error) bool {
 	return strings.Contains(err.Error(), "Terminating state with DeletionTimestamp")
 }
+
+func (r *JenkinsReconciler) setInitialConditions(jenkins *v1alpha2.Jenkins) error {
```
Comment (akram) [inline controllers/jenkins_controller.go]: Not sure what it does: why are we rolling all the conditions here ?

**Review by akram — CHANGES_REQUESTED** [review]

File: config/crd/bases/jenkins.jenkins.io_jenkins.yaml (hunk 2 above)
Comment (waveywaves) [reply config/crd/bases/jenkins.jenkins.io_jenkins.yaml]: +1

**Review by waveywaves — CHANGES_REQUESTED** [review]

File: api/v1alpha2/jenkins_types.go (hunk 1 above)
Comment (jkhelil) [reply api/v1alpha2/jenkins_types.go:477]: yes

File: controllers/jenkins_controller.go (hunk 4 above)
Comment (jkhelil) [reply controllers/jenkins_controller.go:254]: this is harmonized with conditionAvailable ext, so we are not useing names

File: controllers/jenkins_controller.go [hunk 6]
```diff
@@ -58,6 +58,14 @@ const (
 	ConfigMapKind          = "ConfigMap"
 	containerProbeURI      = "login"
 	containerProbePortName = "http"
+
+	reconcileInit             = "Init"
+	reconcileInitMessage      = "Initializing Jenkins operator"
+	reconcileFailed           = "ReconcileFailed"
+	reconcileCompleted        = "ReconcileCompleted"
+	reconcileCompletedMessage = "Reconcile completed successfully"
+
```
Comment (akram) [inline controllers/jenkins_controller.go]: Can you change this to Reconciliation insteand or Reconcile

File: controllers/jenkins_controller.go [hunk 7]
```diff
@@ -58,6 +58,14 @@ const (
...
+	ConditionReconcileComplete conditionsv1.ConditionType = "ReconcileComplete"
```
Comment (akram) [inline controllers/jenkins_controller.go]: and this one also

**Review by akram — CHANGES_REQUESTED** [review]

File: controllers/jenkins_controller.go (hunk 6 above)
Comment (jkhelil) [reply controllers/jenkins_controller.go]: done

File: controllers/jenkins_controller.go (hunk 7 above)
Comment (jkhelil) [reply controllers/jenkins_controller.go]: done

**Pull request merged**